conn.go

```diff
@@ -32,18 +32,125 @@ import (
 // Commands are buffered. Flush sends all buffered commands in a single batch.
 type Conn struct {
 	TestDial nltest.Func // for testing only; passed to nltest.Dial
-	NetNS    int         // Network namespace netlink will interact with.
-	sync.Mutex
-	messages []netlink.Message
-	err      error
+	NetNS    int         // fd referencing the network namespace netlink will interact with.
+
+	mu        sync.Mutex // protects the following state
+	messages  []netlink.Message
+	err       error
+	nlconn    *netlink.Conn // netlink socket using NETLINK_NETFILTER protocol.
+	closeOnce sync.Once
+}
+
+// ConnOption is an option to change the behavior of the nftables Conn returned by Open.
+type ConnOption func(*Conn)
+
+// Open returns a ("long-running") netlink connection for querying and modifying
+// nftables. It should be closed by calling Close() to close the underlying
+// netlink connection and release associated resources and pending operations.
+//
+// Please note that it is not necessary to call Open in order to create an
```

Comment:
I think once a constructor exists, people will be drawn to using it, even if not strictly necessary.

Instead of documenting that a Conn{} can be used, how about we make the long-running connection bit a ConnOption, and have Open() behave the same way as Conn{} now

Reply:
should we then still name it `Open` or rather `New?` And then an `KeepOpen` option?

Reply:
I think `New` would be a slightly better name, yeah.

Reply:
refactored accordingly with `New` and `AsLasting` for requesting a "lasting" connection. Also refactored tests to use the constructor instead of creating `Conn` instances directly.